### PR TITLE
use daml-lf scalacopts for ledger-service

### PR DESCRIPTION
**Pull request by S11001001**
`lf_scalacopts` is meant to reflect, more or less, increases in strictness preferred for DAML runtime, and let us tighten up checks on a portion of the codebase without touching everything all at once. The JSON API and its support code should be included in that set. We do that and solve the additional warnings detected.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/jwt/src/main/scala/com/digitalasset/jwt/Main.scala [hunk 1]
```diff
@@ -32,9 +32,8 @@ object Main {
             e.printStackTrace()
             sys.exit(ErrorCodes.GenerateKeysError)
         }
-      case Some(Config(None, Some(GenerateJwt(Some(publicKey), Some(privateKey))))) =>
-        JwtGenerator.generate(
-          domain.KeyPair(publicKey = Seq.empty[Byte], privateKey = Seq.empty[Byte])) match {
+      case Some(Config(None, Some(GenerateJwt(Some(publicKey @ _), Some(privateKey @ _))))) =>
+        JwtGenerator.generate match {
```
Comment (S11001001) [inline ledger-service/jwt/src/main/scala/com/digitalasset/jwt/Main.scala:36]: @leo-da We're just a little more explicit that the contents of the argument are ignored, now. How does this sound to you?

File: ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala [hunk 2]
```diff
@@ -408,7 +409,7 @@ class WebsocketServiceIntegrationTest
           via parseResp runWith remainingDeltas)
       } yield
         inside(rescan) {
-          case (Vector((fstId, fst), (sndId, snd)), Vector(observeConsumed), Some(_)) =>
+          case (Vector((fstId, fst @ _), (sndId, snd @ _)), Vector(observeConsumed), Some(_)) =>
```
Comment (leo-da) [inline ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala:412]: what is the differnces `fst @ _` vs `fst`. I don't see `fst` used in the scope of this pattern match, can it be be:
```
case (Vector((fstId, _), (sndId, _)), Vector(observeConsumed), Some(_))
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala [hunk 3]
```diff
@@ -321,7 +321,7 @@ object ValuePredicate {
     }
 
     (typ match {
-      case tc @ iface.TypeCon(iface.TypeConName(id), typArgs) =>
+      case tc @ iface.TypeCon(iface.TypeConName(id), typArgs @ _) =>
```
Comment (leo-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala:324]: Why not `case tc @ iface.TypeCon(iface.TypeConName(id), _)`?

File: ledger-service/jwt/src/main/scala/com/digitalasset/jwt/Main.scala (hunk 1 above)
Comment (leo-da) [reply ledger-service/jwt/src/main/scala/com/digitalasset/jwt/Main.scala:36]: I should have removed this `main`... It was supposed to be the tool to generate keys for the JWT, but I don't think we are using it right now.

In any case regarding the `case Some(Config(None, Some(GenerateJwt(Some(publicKey @ _), Some(privateKey @ _)))))`
you can still used `publicKey` and `privateKey` values, right? Nothing prevents you from this. Why not just use `_` instead. But it reads better like `publicKey @ _`, kind of self-documenting.

Comment (S11001001) [reply ledger-service/jwt/src/main/scala/com/digitalasset/jwt/Main.scala:36]: When you don't use a pattern variable, `-Ywarn-unused` suggests this form, for exactly the reason you say.

**Pull request merged**